pkg/runtime/start.go

```diff
@@ -0,0 +1,45 @@
+package runtime
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"golang.org/x/sync/errgroup"
+)
+
+func (r *Runtime) Start() {
+	if r.Service == nil {
+		r.Logger.Error("no Service set")
+		return
+	}
+	if r.SyncImpl == nil || len(r.SyncImpl) == 0 {
```

Comment:
```suggestion
	if len(r.SyncImpl) == 0 {
```
nil slices have length 0